app/base/static/assets/js/vis/simple-bar-chart.js

```diff
@@ -51,46 +57,97 @@ class SimpleBarChart {
         let g = svg.append("g")
             .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
 
-        g.append("g")
-            .attr("class", "axis axis--x")
-            .attr("transform", "translate(0," + height + ")")
-            .call(d3.axisBottom(x)
-              .tickFormat(d3.timeFormat("%Y-%m-%d")).tickValues(x.domain().filter(function (d, i) {
-                 return !(i % 10)
-            })))
-            .selectAll("text")
-            .style("text-anchor", "end")
-            .attr("dx", "-.8em")
-            .attr("dy", ".15em")
-            .attr("transform", "rotate(-65)");
-
-        g.append("g")
-            .attr("class", "axis axis--y")
-            .call(d3.axisLeft(y))
-            .append("text")
-            .attr("class", "axis-title");
-
-        g.selectAll(".bar")
-            .data(data)
-            .enter().append("rect")
-            .attr("x", function (d) {
+        //declare xAxis and xAxisE element variable to be used in resize function
+        let xAxis = d3.axisBottom(x)
+                      .tickFormat(d3.timeFormat("%Y-%m-%d")).tickValues(x.domain().filter(function (d, i) {
+                            return !(i % 10)
+                        }));
+        let xAxisEL=g.append("g")
+                     .call(xAxis);
+
+        xAxisEL.attr("class", "axis axis--x")
+               .attr("transform", "translate(0," + height + ")")
+               .selectAll("text")
+               .style("text-anchor", "end")
+               .attr("dx", "-.8em")
+               .attr("dy", ".15em")
+               .attr("transform", "rotate(-65)");
+
+
+        //declare yAxis and yAxis element variable to be used in resize function
+        let yAxis = d3.axisLeft().scale(y);
+        let yAxisEL=g.append("g")
+            .call(yAxis);
+        yAxisEL.attr("class", "axis axis--y")
+               .append("text")
+               .attr("class", "axis-title");
+
+       //declare bar variable to be used in resize function
+       let bar=g.selectAll(".bar")
+                .data(data)
+                .enter().append("rect");
+
+        bar.attr("x", function (d) {
+            return x(d.index);
+        })
+        .attr("y", function (d) {
+            return y(d[field]);
+        })
+        .attr("width", x.bandwidth())
+        .attr("height", function (d) {
+            return height - y(d[field]);
+        })
+        .attr("fill", '#4682B4')
+        .on("mousemove", function(d){
+            tooltip_barchart
+                .style("left", d3.event.pageX - 50 + "px")
+                .style("top", d3.event.pageY - 70 + "px")
+                .style("display", "inline-block")
+                .html(d.index + '-> ' + d[field]);
+        })
+        .on("mouseout", function(d){ tooltip_barchart.style("display", "none");});
+
+
+        //declare resize function
+        function resize() {
```

Comment:
I noticed that there is also a function called resize() in the simple-line-chart.js. Do they have the same behaviour? If yes, then perhaps they can be refactored and defined as a function in common.js

Reply:
yes, agree, resize() is like a interface abstract method, and the detail is different in different vis chart.